### ISSUE TITLE
Updated Neovim  to 0.5 version on windows and coc-flutter not working

### ISSUE THREAD

**Issue by jithusjacob** [iamcco/coc-flutter#123]
I just installed Neovim 0.5.0 on windows and coc-flutter is not working. Then I installed Neovim 0.4.5 and its working fine now.
Any way it can work for Neovim 0.5.0

**Comment by Kavantix** [comment]
@jithusjacob I’m also using 0.5 so it should work
Could you add the info requested in the template

**Comment by jithusjacob** [comment]
**Describe the bug**
I installed Neovim version 0.5 and used my old init.vim. Coc action and error detecting for flutter was not working.
So installed  neovim 0.4.5 and its working now. This is on windows 10



**Expected behavior**
Code action and error detecting should work for neovim 0.5

**Screenshots**
Neovim 0.4.5
![image](https://user-images.githubusercontent.com/16571206/126596256-6fbbe1c3-46fc-46bd-9b83-0e66aea5f6ce.png)

![image](https://user-images.githubusercontent.com/16571206/126596293-ff93175a-0dc1-402c-8df1-63d0acb3b97c.png)

Neovim 0.5.0
No error and code action
![image](https://user-images.githubusercontent.com/16571206/126600560-1f69708f-a0ef-460e-b498-f5e5f91af9a8.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Vim or Neovim:NeoVim
 - (Neo)vim version:0.5.0

**Output channel:**

1. Set `"flutter.trace.server":   "verbose"`
2. Restart coc using `:CocRestart`
3. Reproduce the issue
4. `:CocCommand workspace.showOutput` open output list and select `flutter`

Paste Log from above:
[dev-command]: register dev command                                                                                                                        [sdk]: No local fvm sdk
[sdk]: which flutter command => C:\Users\jithu\Documents\flutter\bin\flutter.BAT
[fs]: executing command cat C:\Users\jithu\Documents\flutter\version
[sdk]: searchPaths: ~/snap/flutter/common/flutter
[sdk]: which dart command => C:\Users\jithu\Documents\flutter\bin\dart.BAT
[sdk]: dart command path => C:\Users\jithu\Documents\flutter\bin\dart.BAT
[sdk]: dart sdk home => C:\Users\jithu\Documents\flutter
[sdk]: analyzer path => C:\Users\jithu\Documents\flutter\bin\snapshots\analysis_server.dart.snapshot
[sdk]: dart command path => C:\Users\jithu\Documents\flutter\bin\dart.bat
[sdk]: Dart SDK not found!
[sdk]: {
  "dartHome": "C:\\Users\\jithu\\Documents\\flutter",                                                                                                        "analyzerSnapshotPath": "C:\\Users\\jithu\\Documents\\flutter\\bin\\snapshots\\analysis_server.dart.snapshot"
}
[lsp-server]: flutter SDK not found!
[daemon]: got message: {                                                                                                                                     "event": "daemon.connected",
  "params": {
    "version": "0.6.0",
    "pid": 11904
  }
}
[daemon]: got message: {
  "event": "daemon.logMessage",
  "params": {
    "level": "status",
    "message": "Starting device daemon..."
  }
}
[daemon]: got message: {
  "id": 1
}
[daemon]: Message without event
[daemon]: got message: {
  "event": "device.added",
  "params": {
    "id": "chrome",
    "name": "Chrome",
    "platform": "web-javascript",
"emulator": false,
    "category": "web",
    "platformType": "web",
    "ephemeral": false,
    "emulatorId": null
  }
}
[daemon]: New device: Chrome
[daemon]: got message: {
  "event": "device.added",
  "params": {
    "id": "edge",
    "name": "Edge",
    "platform": "web-javascript",
    "emulator": false,
    "category": "web",
    "platformType": "web",
    "ephemeral": false,
    "emulatorId": null
  }
}
[daemon]: New device: Edge

**Config**
none

**Comment by Kavantix** [comment]
Which version of flutter are you using?
And can you check if `C:\Users\jithu\Documents\flutter\bin\dart.BAT` exists

**Comment by Kavantix** [comment]
Also does this file exist: `C:\Users\jithu\Documents\flutter\version`?

**Comment by jithusjacob** [comment]
Flutter 2.2.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f4abaa0735 (3 weeks ago) • 2021-07-01 12:46:11 -0700
Engine • revision 241c87ad80
Tools • Dart 2.13.4

and yes C:\Users\jithu\Documents\flutter\bin\dart.BAT exists
![image](https://user-images.githubusercontent.com/16571206/126603078-d13090e2-b78b-4e00-8897-f74a5bb7cd9c.png)

**Comment by jithusjacob** [comment]
> C:\Users\jithu\Documents\flutter\version

yes it exists

**Comment by Kavantix** [comment]
@jithusjacob  Could you try if #124 fixes it?

**Comment by jithusjacob** [comment]
How do I try this update. Can you  give the steps to install this change?

**Comment by Kavantix** [comment]
You can add the following to your vim config (if you use plug and have node installed)
```vim
 Plug 'iamcco/coc-flutter', {'do': 'npm install'}
```

**Comment by Kavantix** [comment]
You can add the following to your vim config (if you use plug and have node installed)
```vim
 Plug 'iamcco/coc-flutter', {'branch': 'fix/flutter-version-windows',  'do': 'npm install'}
```

**Comment by Kavantix** [comment]
You probably also need to uninstall coc-flutter using `:CocUninstall coc-flutter` otherwise it will still pick that one

**Comment by jithusjacob** [comment]
.Not Sure about the install.I usually install using init.vim having below lines
" COC PLUGIN SETUP:
  7 let g:coc_global_extensions = [
  6   \ 'coc-flutter',
  5   \ 'coc-snippets',
  4   \ 'coc-yaml',
  3   \ 'coc-explorer',
  2   \ 'coc-markdownlint',
  1   \]

So deleted coc-flutter from init.vim and did :CocUninstall coc-flutter. Then added the plugin and installed but getting the below error while installing.Now the Coc-flutter is not starting itself
![image](https://user-images.githubusercontent.com/16571206/126611392-6f3117de-f8ad-477a-b6e0-50f8ec96f330.png)

**Comment by Kavantix** [comment]
Can you manually run `npm install` in `C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter`

**Comment by jithusjacob** [comment]
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter> npm install

> coc-flutter@1.9.2 prepare
> husky install && npm run clean && npm run build

husky - Git hooks installed

> coc-flutter@1.9.2 clean
> rm -rf ./out

'rm' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code 1
npm ERR! path C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c husky install && npm run clean && npm run build

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\jithu\AppData\Local\npm-cache\_logs\2021-07-22T08_47_28_285Z-debug.log
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter> nvim C:\Users\jithu\AppData\Local\npm-cache\_logs\2021-07-22T08_47_28_285Z-debug.log

**Comment by Kavantix** [comment]
Ah I see whats wrong, you can change line 23 of the package json to:
```json
    "prepare": "husky install && npm run build"
```
That will prevent the clean command from running since it was not setup for windows it seems

**Comment by Kavantix** [comment]
@iamcco can you look into fixing the `npm run clean` on windows

**Comment by iamcco** [comment]
> @iamcco can you look into fixing the `npm run clean` on windows

fix the clean script on windows https://github.com/iamcco/coc-flutter/pull/125/files

**Comment by jithusjacob** [comment]
> Ah I see whats wrong, you can change line 23 of the package json to:
> 
> ```json
>     "prepare": "husky install && npm run build"
> ```
> 
> That will prevent the clean command from running since it was not setup for windows it seems

I don't know much about npm. I just tried npm init and created package json in that added the prepare statement and ran npm install again.But getting same error

**Comment by Kavantix** [comment]
The command should be fixed now.
Can you run `git reset --hard` then `git pull --rebase` and then `npm install` again. (all in `C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter`)

**Comment by jithusjacob** [comment]
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter> git reset --hard
HEAD is now at 12eb6e8 fix(version-windows): Fixed reading flutter version on windows
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter> git pull --rebase
Already up to date.
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter> npm install

> coc-flutter@1.9.2 prepare
> husky install && npm run clean && npm run build

husky - Git hooks installed

> coc-flutter@1.9.2 clean
> ./node_modules/.bin/rimraf ./out

'.' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code 1
npm ERR! path C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c husky install && npm run clean && npm run build

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\jithu\AppData\Local\npm-cache\_logs\2021-07-22T11_24_20_208Z-debug.log
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter>

**Comment by Kavantix** [comment]
First of all, thanks for helping us test this! We don't use windows.
I made another change to the script can you again try pulling and running npm install

**Comment by jithusjacob** [comment]
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter> npm install

> coc-flutter@1.9.2 prepare
> husky install && npm run clean && npm run build

husky - Git hooks installed

> coc-flutter@1.9.2 clean
> rimraf ./out


> coc-flutter@1.9.2 build
> NODE_ENV=production node esbuild.js

'NODE_ENV' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code 1
npm ERR! path C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c husky install && npm run clean && npm run build

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\jithu\AppData\Local\npm-cache\_logs\2021-07-22T11_32_48_862Z-debug.log
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter>

**Comment by Kavantix** [comment]
Alright, we are getting further at least.
@iamcco could you again look this error.

@jithusjacob to get you a step further now, can you run `node esbuild.js` directly instead of `npm run install`

**Comment by jithusjacob** [comment]
PS C:\Users\jithu\AppData\Local\nvim\plugged\coc-flutter> node esbuild.js
build succeeded

**Comment by Kavantix** [comment]
Cool, test it now :)

**Comment by jithusjacob** [comment]
Super its working perfectly fine as in Neovim 0.4.5. thank so much for the support and helping me step by step. Just one doubt.

Will need to use the plugin or I can use it as COC-extension now

**Comment by Kavantix** [comment]
Good to hear.
Once @iamcco has reviewed my changes we'll publish it and then you can remove the line from your vim config and install it using the regular channels again.

**Comment by jithusjacob** [comment]
Thanks

**Comment by iamcco** [comment]
> Alright, we are getting further at least.
> @iamcco could you again look this error.
> 
> @jithusjacob to get you a step further now, can you run `node esbuild.js` directly instead of `npm run install`

will look into it later

**Comment by Kavantix** [comment]
@jithusjacob I just published v1.9.3 so you should be able to switch back to the normal version.
Thanks again for helping us test!

**Comment by jithusjacob** [comment]
@Kavantix Thanks for extended help..... switched back to normal version and it working fine.
